WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerSearchView.java

```diff
@@ -64,9 +63,7 @@ public boolean onMenuItemActionCollapse(MenuItem item) {
     public void enableSearchMode() {
         mMenuEdit.setVisible(false);
         mSitePickerActivity.setIsInSearchModeAndNullifyAdapter(true);
-        SitePickerSearchAdapter adapter = (SitePickerSearchAdapter) mSitePickerActivity.getAdapter();
-        mSitePickerActivity.getRecycleView().swapAdapter(adapter, true);
-        adapter.loadSites();
+        mSitePickerActivity.getRecycleView().swapAdapter(mSitePickerActivity.getAdapter(), true);
```

Comment:
Sort of related to the above comment, right now both `SitePickerActivity` and `SitePickerAdapter` have their own `mIsInSearchMode` and `mLastSearch` variables, which seems error prone. I think these can be handled solely by the adapter.